### DISCUSSION TITLE
passing in list of objects, and have rule test each object.

### DISCUSSION THREAD

**Discussion by tisdalestar6git** [microsoft/RulesEngine#169] [Q&A]
I can pass in a single object to a rule. But has anyone used this engine to pass a list of objects, and have a rule run on each object in the list?  Or  should is just go though my list and call the rule for each object?

**Comment by ghost** [comment]
I have the same question.  Has anyone come across this or have insight into the approach?

**Comment by asulwer** [comment]
the demo app uses a list

[NestedInputDemo](https://github.com/microsoft/RulesEngine/blob/main/demo/DemoApp/NestedInputDemo.cs)

**Reply by tisdalestar6git** [reply]
Thanks for your reply, however I'm still not sure how to setup a rule for a list of objects. The demo is referencing listprop[X]. In a real application this is not usable. Can you write a rule that would have the rule engine iterate over a list of objects?
